src/engine.rs

```diff
@@ -666,6 +666,13 @@ impl Reedline {
         self.painter.paint_line(msg)
     }
 
+    /// <placeholder docs>
```

Comment:
I'll update these docs once the direction reaches a consensus (that we will implement the feature in this way).

The current patch can fit my requirement, but I can lack some of the design principles or other considerations within this project's practice. So please let me know if there is anything should be considered. And if it's good to proceed, please tell me explicitly also.